Libraries/DirectMLX.h

```diff
@@ -2342,7 +2341,14 @@ namespace dml
             axisSizeSum += outputAxisSize;
         }
 
+#if defined(_DEBUG)
```

Comment:
`#ifndef NDEBUG` would be better here, since NDEBUG corresponds for C-style assertions, and `_DEBUG` is a non-standard macro defined by Visual Studio, but not necessarily other compilers. @adtsai - does the rest look good to you?